UI / UX changes : closes #41

lukaqueres opened the pull request:
Changed ticker input validation method to regex.

Added loading popup

mrhappyasthma commented:
Thanks for the PRs :)

[copy/pasted from my comment on the other thread:]

Would it be possible to replace the tabs back to 2x spaces indentation? It's hard to isolate the actual changes in the diff since every line is modified.

(I'm open to changing the amount of spacing, but if possible let's isolate that. Seems like the Google styleguide recommends 4x spaces, which I also wasn't adhering to: https://google.github.io/styleguide/pyguide.html#34-indentation)

lukaqueres commented:
I can correct it, somehow I made it so that in this PR is also the previous one, fixing errors `undefined`.

lukaqueres commented:
Sorry for so many problems, damn indentations

mrhappyasthma commented on templates/searchbox.html:
```diff
@@ -7,28 +7,26 @@
-  };
+document.querySelector("#ticker").addEventListener("keypress", (event) => {
+  let key = event.key;
+  if (!key.match(/[A-Za-z]/)) {
```
Doesn't this also need to be able to match periods? For tickers like `BRK.B`?

(We have some inconsistencies in the current code, even. Ideally we'd also accept formats like `BRK-B`. But some of the backends are particular about which one they use. And we don't currently sanitize the input for each backend to make sure it's in the accepted format. That's a separate bug but one we can also fix up to improve things)

mrhappyasthma replied:
Fixed in the latest version. Thanks!

mrhappyasthma commented on src/YahooFinance.py:
```diff
@@ -9,12 +9,12 @@ class YahooFinanceQuote:
   # Expects the ticker symbol as the only argument.
```
It doesn't look like we have any real changes in this file. Can we just revert it back to HEAD?

mrhappyasthma commented on src/DataFetcher.py:
```diff
@@ -8,6 +9,9 @@
 from src.YahooFinance import YahooFinanceQuoteSummary, YahooFinanceQuoteSummaryModule
 from threading import Lock
 
+logger = logging.getLogger("IsThisStockGood")
```
Can we revert this file (except for the new Logger lines: 1, 12, 240, 242, 244?) to eliminate all the other whitespace changes?

(it may be simpler to revert to HEAD and just re-add those few lines back)

mrhappyasthma commented:
This looks awesome! Thank you for adding this.

lukaqueres commented:
All the pleasure is mine, glad to help :D

lukaqueres commented:
Fixed Regex, should now support `-`, `,` and `.`, I have very little expirience with tickers, so I can only count on feedback.

lukaqueres committed:
New ticker input validation using regex, new loading popup

1. Allow usage of `Enter` button along with other functional buttons ( `tab`, `arrows` ) in input,
2. Add support for dot, colon and hyphen in input,
3. Add custom loading element with animation to show loading process.

lukaqueres commented:
So, hopefully now it will be ok 👍

mrhappyasthma commented:
Looks good!
